usersimcrs/simulator/agenda_based/interaction_model.py

```diff
@@ -116,19 +119,96 @@ def initialize_transition_matrices(
         The compound intent transition matrix will be:
         {GREETING, REQUEST} -> INFORM : 1
 
+        Note that the compound intent may also include single intents, in case
+        an utterance has a single dialogue act.
+
         Args:
             annotated_conversations: Annotated conversations.
 
         Returns:
             Transition matrices.
         """
-        transition_matrix_single = pd.DataFrame()
-        transition_matrix_compound = pd.DataFrame()
+        single_intent_distribution: DefaultDict[
+            str, DefaultDict[str, int]
+        ] = defaultdict(lambda: defaultdict(int))
+        compound_intent_distribution: DefaultDict[
+            str, DefaultDict[str, int]
+        ] = defaultdict(lambda: defaultdict(int))
+
+        agent_user_interactions = self._get_agent_user_interactions(
+            annotated_conversations
+        )
+        for agent_dialogue_acts, user_dialogue_acts in agent_user_interactions:
+            compound_agent_intent = "_".join(
```

Comment:
Perhaps mention here or in the method-level docstring that the key/ID for compond intent is the concatenation of individual intents, alphabetically sorted, as we're treating them as a set.

Reply:
Done.